src/python/WMCore/Storage/TrivialFileCatalog.py

```diff
@@ -196,55 +232,144 @@ def tfcFilename(contactString):
+                                #now replace anything inside () with $1, for example (.*) --> $1, (store/.*) --> $1
+                                result = re.sub('\(.*\)','$1',result)
+                                tfcInstance.addMapping(str(proc['protocol']), str(match), str(result), chain, 'pfn-to-lfn')
+    
+    return tfcInstance
 
+def rseName(storageAttr):
+    rse = None
+    catalog = getCatalog(storageAttr)
+    storageJsonName = tfcFilename(catalog)
     try:
-        node = xmlFileToNode(filename)
+        jsonFile=open(storageJsonName,encoding="utf-8")
+        jsElements = json.load(jsonFile)
     except Exception as ex:
-        msg = "Error reading TrivialFileCatalog: %s\n" % filename
+        msg = "TrivialFileCatalog.py:rseName() Error reading FileCatalog: %s\n" % storageJsonName 
         msg += str(ex)
         raise RuntimeError(msg)
-
-    parsedResult = nodeReader(node)
-
-    tfcInstance = TrivialFileCatalog()
-    for mapping in ['lfn-to-pfn', 'pfn-to-lfn']:
-        for entry in parsedResult[mapping]:
-            protocol = entry.get("protocol", None)
-            match = entry.get("path-match", None)
-            result = entry.get("result", None)
-            chain = entry.get("chain", None)
-            if True in (protocol, match == None):
-                continue
-            tfcInstance.addMapping(str(protocol), str(match), str(result), chain, mapping)
-    return tfcInstance
-
-
-def loadTFC(contactString):
+    for jsElement in jsElements:
+        if jsElement['site'] == storageAttr['storageSite'] and jsElement['volume'] == storageAttr['volume']:
+            rse = jsElement['rse']
+            break
+    return rse
+
+def lfnPrefix(storageAttr):
+    catalog = getCatalog(storageAttr)
+    #now get lfn prefix
+    storageJsonName = tfcFilename(catalog)
+    try:
+        jsonFile = open(storageJsonName,encoding="utf-8")
+        jsElements = json.load(jsonFile)
+    except Exception as ex:
+        msg = "TrivialFileCatalog.py:lfnPrefix() Error reading FileCatalog: %s\n" % storageJsonName 
+        msg += str(ex)
+        raise RuntimeError(msg)
+    #now loop over elements, select the right one and fill lfn-to-pfn
+    lfnPrefixCol = []
+    for jsElement in jsElements:
+        if jsElement['site'] == storageAttr['storageSite'] and jsElement['volume'] == storageAttr['volume']: 
+            #now loop over protocols
+            for proc in jsElement['protocols']:
+                #check found match
+                if proc['protocol'] == storageAttr['protocol']:
+                    #check if prefix in protocol block
+                    if 'prefix' in proc.keys():
+                        lfnPrefixCol.append(proc['prefix'])
+                    #here is rules
+                    else:
+                        #loop over rules and keep all of them
+                        for rule in proc['rules']:
+                            lfnPrefixCol.append(rule['pfn'].replace('/$1','').replace('$1',''))
```

Comment:
Is this line required to keep the same compatibility as expected in the job runtime (required by CMSSW?)?